src/dev/flang/be/interpreter2/LValue.java

```diff
@@ -0,0 +1,352 @@
+   *
+   * @param c the clazz of the value, for debugging only
+   *
+   * @param off the offset of the value within this
+   *
+   * @return the LValue to rev
+   */
+  public LValue at(Clazz c, int off)
+  {
+    return new LValue(c,
+                      container,
+                      offset + off);
+  }
+
+
+  /**
+   * Store this value in a field
+   *
+   * @param slot the slot that addresses the field this should be stored in.
+   *
+   * @param size the size of the data to be stored
+   */
+  void storeNonRef(LValue slot, int size)
+  {
+    if (PRECONDITIONS) require
+      (size == Layout.get(_clazz).size());
+
+    container.storeNonRef(slot, size, offset);
+  }
+
+
+  /**
+   * Does this value equal the value in slot of given size on a low-level
+   * bit-wise comparison?
+   *
+   * @param slot the slot that addresses the field this should be compared
+   * against.
+   *
+   * @param size the size of the data to be compared.
+   */
+  boolean equalsBitWise(LValue slot, int size)
+  {
+    return container.equalsBitWise(slot, size, offset);
+  }
+
+
+  /**
+   * Debugging only: Check that this value is valid as the current instance for
+   * a feature with given static clazz.
+   *
+   * @param expected the static clazz of the feature this value is called on.
+   *
+   * @throws Error in case this does not match the expected clazz
+   */
+  void checkStaticClazz(Clazz expected)
+  {
+    if (expected.isRef())
+      {
+        throw new Error("LValue (" + _clazz + " not allowed for dynamic clazz " + expected);
+      }
+    if (expected != _clazz)
+      {
+        throw new Error("Runtime clazz "+_clazz+" does not equal static "+expected);
+      }
+  }
+
+
+  /**
+   * Return the instance this value contains.  If this is an Instance, return
+   * this, if this is an LValue containing an instance, get that instance.
+   */
+  Instance instance()
+  {
+    if (PRECONDITIONS) require
+      (_clazz.isRef());
+
+    return (Instance) container.refs[offset];
+  }
+
+
+  /**
+   * Return the ArrayData this value contains.  If this is an ArrayData, return
+   * this, if this is an LValue containing an ArrayData, get that ArrayData.
+   */
+  ArrayData arrayData()
+  {
+    return (ArrayData) container.refs[offset];
+  }
+
+
+  /**
+   * Return the tag of this choice.
+   */
+  public int tag()
+  {
+    if (PRECONDITIONS) require
+      (_clazz.isChoice() & !_clazz.isChoiceOfOnlyRefs());
+
+    var tag = container.nonrefs[offset];
+    if (POSTCONDITIONS) ensure
+      (tag >= 0);
+
+    return tag;
+  }
+
+
+  /**
+   * toString
+   *
+   * @return
+   */
+  public String toString()
+  {
+    return "lvalue[" + container + "@" + offset + "(" + _clazz + ")]" +
+      (_clazz == Clazzes.i32.getIfCreated() ? " (" + i32Value() + ")" :
+       _clazz == Clazzes.u32.getIfCreated() ? " (" + u32Value() + ")" :
+       _clazz == Clazzes.u8.getIfCreated() ? " (" + u8Value() + ")" :
```

Comment:
You have added `u8` here. Might also make sense to add `i8`, `i16`, `u16`, `f32`, `f64`.